### DISCUSSION TITLE
Performance Tuning

### DISCUSSION THREAD

**Discussion by Muraligowtham** [apache/pulsar#19277] [Q&A]
Hi All, We have a pulsar cluster created with 1 broker(10vpcu, 32gb memory, 30g jvm), 1 bookeeper(10vpcu, 32gb memory, 30g jvm), 1 proxy (10vpcu, 32gb memory, 30g jvm),and 1 zookeeper(6vpcu, 12gb memory, 11g jvm). With this set up, we ran a performance test using a producer POC application that runs on our 10 ECS tasks.
Test scenario: sending 3kb json payload to the same non partitioned topic (under public/default namespace) from all 10 ecs tasks (1 producer per ecs task)

Throughput result: 2500mps ingested and 2500mps consumed. This is the max that we were able to produce. does this look right or are we expecting a better perfomance in this scenario? can any one pls help on tuning this performance more

on the other hand when we created one producer object per message(which is bad and wrong), we surprisingly saw better performance. any reasons why it couldn't have performed well.

Note: we did not use batching here

**Comment by syk-coder** [comment]
In the second scenario, as you are using just one producer, the batching will become very effective and would increase the overall throughput. You can try tuning the client batching configurations to check further. 
Link to the docs:  https://pulsar.apache.org/reference/#/next/client/client-configuration-producer ( I Cannot exactly without more info on your client configs, but I think, increasing the [batchingMaxPublishDelayMicros](https://pulsar.apache.org/reference/#/next/client/client-configuration-producer?id=batchingmaxpublishdelaymicros) might help you in this case). This is similar to the lingerMs of Kafka
